magento/pwa-studio
# Preselect lowest cost shipping method for authed users without one

## Description

This PR pre-selects the lowest cost shipping method for authenticated (signed in) users who don't yet have a saved / preferred shipping method. 

I also added a new `INITIALIZING` value to the `displayState` enumeration to remove a visual hiccup that occurred during auto-selection of a shipping method. The `ShippingMethod` component should be `INITIALIZING` before fetching its data and while the auto-selection is in progress.

Adding this `INITIALIZING` state allowed us to greatly simplify downstream logic as well 🎉 .

## Related Issue





Closes PWA-573.

## Acceptance 



### Verification Stakeholders


@soumya-ashok 

### Specification



### Verification Steps


1. Sign in
1. Add an item to your cart
1. Set a shipping address (via mini-cart if #2380 isn't merged yet)
1. Go to the `/checkout` page
1. Verify that you're never actually presented with the shipping radios to select; the lowest cost shipping option (FREE) is selected for you (after some loading) and you are taken directly to the payment section
1. Fill out the payment section
1. Verify that you can successfully place your order

## Screenshots / Screen Captures (if appropriate)

![573_update](https://user-images.githubusercontent.com/13182778/82248231-6537c600-990d-11ea-942c-bf7d9b1fda1e.gif)

## Checklist

* I have updated the documentation accordingly, if necessary.
* I have added tests to cover my changes, if necessary.
